packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,199 @@
+| onClose            | `Callback`                                                               | `undefined` | Called when tooltip is closed                                                                                                                            |          |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic example
+<Tooltip content="Amount reversed to customer bank account" placement="top">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>
+```
+
+### Working with non-interactive triggers
+
+To make tooltip apear on hovering over non-interactive elements such as icons, badges, counters etc. We will provide `shouldWrapChildren` prop which will wrap the trigger component in a interactive div element with all the trigger props
+
+```js
+// non-interactive element as trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
+  <Counter value={100} />
+</Tooltip>
+```
+
+> Note: If users don't wrap the non-interactive elements the tooltip won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
+
+To make custom elements work the components needs to expose:
+
+- ref
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+```jsx
+import type { TooltipTriggerProps } from "@razorpay/blade/components";
+
+type MyCustomButtonProps = {} & TooltipTriggerProps
+const MyCustomButton = React.forwardRef(...);
+
+// custom trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
+  <MyCustomButton>Click me<MyCustomButton>
+</Tooltip>
+```
+
+## Library
+
+We will be using [FloatingUI](https://floating-ui.com/) to position the tooltip & handle the basic tooltip logic. Floating UI supports both Web & ReactNative.
+
+### Bundle Size
+
+With all the neccessary middlewares and features included for our usecase FloatingUI will be around **10kb** gzip.
+
+Check the [BundleJS Analysis](https://bundlejs.com/?q=%40floating-ui%2Freact&treeshake=%5B%7B%0A++arrow%2Cflip%2CFloatingArrow%2Coffset%2CuseFloating%2CuseFocus%2CuseHover%2CuseInteractions%2CuseTransitionStyles%2C%7D%5D&config=%7B%22esbuild%22%3A%7B%22external%22%3A%5B%22react%22%2C%22react-dom%22%5D%7D%7D)
+
+This is worth the size becaue:
+
+- FloatingUI supports both react / react-native
+- It handles
+  - Animations (mount/unmount)
+  - Automatic collision aware placements
+  - Interactions (hover, focus)
+  - Tooltip Arrow
+  - Middlewares
+
+<details>
+  <summary>Implementation detail nuances:</summary>
+
+To make FloatingUI work with Blade components seamlessly there are few things we need to modify:
+
+1. Expose interaction props in all the trigger components
+
+- Web:
+  - onBlur
```

Comment:
so now all our consumers will have access to all these methods?

Reply:
Yes atleast in components which we are promoting as "triggers"